### DISCUSSION TITLE
Self hosted gitlab

### DISCUSSION THREAD

**Discussion by patsevanton** [terrateamio/terrateam#101] [Q&A] [answered]
does terrateam work with self hosted gitlab?

**Comment by bender2352** [comment]
Hi @patsevanton. Apologies for the delayed response. We're still getting used to GitHub Discussions!

GitLab has always been on our radar but we currently do not have a version supporting it. Having said that, we actually just discussed this yesterday and are considering supporting GitLab in 2025. These are just initial conversations and no decisions have been made. Additionally, we're not very familiar with the GitLab interfaces so it's unclear how long development would even take if we decide to move forward. The good news is that we designed Terrateam in such a way that all of the GitHub interfaces can be changed out for other VCS providers. Regardless, it would still be a project to get done.

**Comment by bender2352** [comment, marked as answer]
We've decided to move forward with GitLab and have already started working on the integration.

#150
